core/src/apps/management/recovery_device/homescreen.py

```
@@ -313,15 +305,29 @@ async def _finish_recovery(secret: bytes, backup_type: BackupType) -> Success:
-    else:
-        secret, share = recover.process_slip39(words)
+    try:
+        if not is_slip39:  # BIP-39
+            secret: bytes | None = recover.process_bip39(words)
+        else:
+            secret, share = recover.process_slip39(words)
+    except MnemonicError:
+        from trezor.ui.layouts.recovery import show_invalid_mnemonic
+
+        await show_invalid_mnemonic(word_count)
+        raise _RetryEntry
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)